### PR TITLE
Jane can update her Consent Preferences (caregiver, caregiver actions…

### PR DESCRIPTION
**Pull request by jon-oblander**
…, test results) #172827030

https://www.pivotaltracker.com/n/projects/2448163/stories/172827030

- updated consents to accept audience
- ui validation / error handling updates

### CODE REVIEW

File: bxhealth/src/pages/profile-settings/caregiver-access.js [hunk 1]
```diff
@@ -97,61 +106,73 @@ class CaregiverAccess extends React.Component {
...
+              this.setState({ recordConsentId: response._embedded.consents[0].id });
+            }
+          }
+        });
       
-    }).catch(err => {
-      console.log('Get consent record error:', err);
-    });
+      }).catch(err => {
+        console.log('Get consent record error:', err);
```
Comment (michaelsanchez-pingidentity) [inline bxhealth/src/pages/profile-settings/caregiver-access.js:173]: Minor change, but if we are using console logs for our error handling, let's make sure it's the appropriate log type so it's highlighted/icon accordingly; console.log, .warn, .error, .info. I"ll make sure this is in dev standards doc.

File: bxhealth/src/pages/profile-settings/caregiver-access.js [hunk 2]
```diff
@@ -198,41 +219,84 @@ class CaregiverAccess extends React.Component {
...
+    if (action === 'save') {
+
+      newActionsConsentId = '';
+      newRecordConsentId = '';
+      caregiverUid = this.state.selectedCaregiver.uid;
+
+      // Revoke consents for previous caregiver. 
+      if (this.state.actionsConsentId.length) {
+        this.user.updateUserConsent(this.state.token, {}, this.state.actionsConsentId, 'revoked')
+        .catch(err => {console.log('Error revoking caretaker actions consent', err)});
```
Comment (michaelsanchez-pingidentity) [inline bxhealth/src/pages/profile-settings/caregiver-access.js:236]: Same as other comment. There's a few of these.

**Review by michaelsanchez-pingidentity — CHANGES_REQUESTED** [review]
Minor, non-logic changes around logging requested.

File: bxhealth/src/pages/profile-settings/caregiver-access.js (hunk 1 above)
Comment (jon-oblander) [reply bxhealth/src/pages/profile-settings/caregiver-access.js]: Good catch. Updating ...

File: bxhealth/src/pages/profile-settings/caregiver-access.js (hunk 2 above)
Comment (jon-oblander) [reply bxhealth/src/pages/profile-settings/caregiver-access.js]: updated.

**Review by michaelsanchez-pingidentity — APPROVED** [review]
Reviewed. Logging changes not worth holding up QA. change later.

**Pull request merged**